packages/agreement-process/test/utils.ts

```diff
@@ -265,3 +269,25 @@ export const getRandomPastStamp = (userId: UserId): AgreementStamp => ({
   who: userId,
   when: subDays(new Date(), randomInt(10)),
 });
+
+export const addSomeRandomDelegations = async (
+  agreement: Agreement
+): Promise<void> => {
+  const states = [delegationState.rejected, delegationState.revoked];
+  const kinds = [
+    delegationKind.delegatedProducer,
+    delegationKind.delegatedConsumer,
+  ];
+
+  for (const state of states) {
+    for (const kind of kinds) {
```

Comment:
I've replaced the `forEach` loop with a `for const` loop because `forEach` does not handle `async/await` correctly.

With this implementation each `addOneDelegation` call should be awaited correctly, while also dynamically generating every combination of `delegationState`/`delegationKind` without having to hardcode those